_security/access-control/document-level-security.md

```diff
@@ -36,6 +36,568 @@ This query specifies that for the role to have access to a document, its `genres
+
+## DLS and multiple roles
+
+OpenSearch combines all DLS queries with the logical `OR` operator. However, when a role that uses DLS is combined with another security role that doesn't use DLS, the query results are filtered to display only documents matching the DLS from the first role. This filter rule also applies to roles that do not grant read documents.
+
+### When to enable `plugins.security.dfm_empty_overrides_all`
+
+When to enable the `plugins.security.dfm_empty_overrides_all` setting depends on whether you want to restrict user access to documents without DLS.
+
+
+To ensure access is not restricted, you can set the following configuration in `opensearch.yml`:
+
+```
+plugins.security.dfm_empty_overrides_all: true
+```
+{% include copy.html %}
+
+
+The following examples show what level of access roles with DLS enabled and without DLS enabled, depending on the interaction. These examples can help you decide when to enable the `plugins.security.dfm_empty_overrides_all` setting.
+
+#### Example: Document access
+
+This example demonstrates that enabling `plugins.security.dfm_empty_overrides_all` is beneficial in scenarios where you need specific users to have unrestricted access to documents despite being part of a broader group with restricted access.
+
+**Role A with DLS**: This role is granted to a broad group of users and includes DLS to restrict access to specific documents, as shown in the following permission set:
+
+```
+{
+  "index_permissions": [
+    {
+      "index_patterns": ["example-index"],
+      "dls": "[.. some DLS here ..]",
+      "allowed_actions": ["indices:data/read/search"]
+    }
+  ]
+}
+```
+
+**Role B without DLS:** This role is specifically granted to certain users, such as administrators, and does not include DLS, as shown in the following permission set:
+
+```
+{
+  "index_permissions" : [
+    {
+      "index_patterns" : ["*"],
+      "allowed_actions" : ["indices:data/read/search"]
+    }
+  ]
+}
+```
+{% include copy.html %}
+
+Setting `plugins.security.dfm_empty_overrides_all` to `true` ensures that administrators assigned Role B can override any DLS restrictions imposed by Role A. This allows specific Role B users to access all documents, regardless of the restrictions applied by Role A's DLS restrictions.
+
+#### Example: Search template access
+
+In this example, two roles are defined, one with DLS and another without DLS, granting access to search templates:
+
+**Role A with DLS:**
+
+```
+{
+  "index_permissions": [
+    {
+      "index_patterns": [
+        "example-index"
+      ],
+      "dls": "[.. some DLS here ..]",
+      "allowed_actions": [
+        "indices:data/read/search",
+      ]
+    }
+  ]
+}
+```
+{% include copy.html %}
+
+**Role B, without DLS**, which only grants access to search templates:
+
+```
+{
+  "index_permissions" : [
+    {
+      "index_patterns" : [ "*" ],
+      "allowed_actions" : [ "indices:data/read/search/template" ]---
```

Comment:
This looks like the formatting is weird?

Reply:
A copy and paste error maybe @leanneeliatra?

Reply:
Indeed, thanks